assignments/this.js

```diff
@@ -10,17 +10,53 @@
 */
 
 // Principle 1
-
+console.log(this);
 // code example for Window Binding
 
 // Principle 2
 
 // code example for Implicit Binding
-
+myObj = {
+    name: "Ashley",
+    age: 27,
+    speak: function (){
+        return `Hi, I'm ${this.name}!`
+    }
+}
 // Principle 3
 
 // code example for New Binding
+function FamilyGuyChar(name){
+    this.name = name;
+    this.speak = function(){
+        return `Hi, my name is ${this.name}.`
+    }
+}
+
+const stewie = new FamilyGuyChar(`Stewie`);
+const meg = new FamilyGuyChar(`Meg`);
+const bryan = new FamilyGuyChar(`Bryan`);
+const peter = new FamilyGuyChar(`Peter`);
+const lois = new FamilyGuyChar(`Lois`);
+const chris = new FamilyGuyChar(`Chris`);
+
+console.log(stewie.speak());
+console.log(meg.speak());
+console.log(bryan.speak());
+console.log(peter.speak());
+console.log(lois.speak());
+console.log(chris.speak());
 
 // Principle 4
 
-// code example for Explicit Binding
\ No newline at end of file
+// code example for Explicit Binding
+
+const skills = ["HTML", "SQL", "CSS", "JS"];
+
+const person = { 
+    name: "Ashley"
+}
+
+function call(skills) {
```

Comment:
You dont have to create a function called `call`. It is actually already built it to the function so in this case since you made the call function, it would look like this: `call.call(Parent, props)`